### PR TITLE
Infomax3D support for finetuning

### PR DESCRIPTION
**Pull request by arunppsg**
## Description

Added support for infomax3d finetuning.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/gnn3d.py [hunk 1]
```diff
@@ -559,27 +585,43 @@ def loss_func(self, inputs, labels, weights):
...
-        #     [torch.cat(conf, dim=0) for conf in encodings3d], dim=0)
-        # NOTE: The above one is logically correct but the below method is similar
-        # to original implementation here: https://github.com/HannesStark/3DInfomax/blob/5cd32629c690e119bcae8726acedefdb0aa037fc/trainer/self_supervised_trainer.py#L26
-        # The different between above and below is that the below one
-        # cannot handle multiple conformers.
-        encodings2d = self.components['2d'](inputs)
-        encodings3d = self.components['3d'](inputs)
-        loss = self.criterion(encodings2d, encodings3d)
+        if self.task == 'pretraining':
+            # encodings2d = []
```
Comment (rbharath) [inline deepchem/models/torch_models/gnn3d.py:589]: Can we remove the commented out code or do we need it still?

**Review by rbharath — COMMENTED** [review]
Have a question, and I think there may be failing tests. We should expect to see 22 tests pass

File: deepchem/models/torch_models/gnn3d.py (hunk 1 above)
Comment (arunppsg) [reply deepchem/models/torch_models/gnn3d.py]: That was a leftover piece of code - I removed it as it is no longer required.

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**